### PR TITLE
feat(gaussdb/opengauss): support new payment for instance resource

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The opengauss instance supports prepaid payment now, by following parameters:
- charging_mode
- period_unit
- period
- auto_renew (update is support)

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support new payment for instance resource.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccOpenGaussInstance_prepaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccOpenGaussInstance_prepaid -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_prepaid
=== PAUSE TestAccOpenGaussInstance_prepaid
=== CONT  TestAccOpenGaussInstance_prepaid
--- PASS: TestAccOpenGaussInstance_prepaid (4964.38s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   4964.463s
```

### CODE REVIEW

**Comment by ShiChangkuo** [comment]
@Lance52259 suggest to just keep commit [8d48c7d](https://github.com/huaweicloud/terraform-provider-huaweicloud/pull/2406/commits/8d48c7da3842fb146a2ba0db97544edb6d0fdd73) in this PR.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccOpenGaussInstance_prepaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccOpenGaussInstance_prepaid -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_prepaid
=== PAUSE TestAccOpenGaussInstance_prepaid
=== CONT  TestAccOpenGaussInstance_prepaid
--- PASS: TestAccOpenGaussInstance_prepaid (4799.27s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   4799.347s
```

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccOpenGaussInstance_prepaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccOpenGaussInstance_prepaid -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_prepaid
=== PAUSE TestAccOpenGaussInstance_prepaid
=== CONT  TestAccOpenGaussInstance_prepaid
--- PASS: TestAccOpenGaussInstance_prepaid (4988.90s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb 4988.994s
```

File: huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance_test.go [hunk 1]
```diff
@@ -79,6 +79,64 @@ func TestAccOpenGaussInstance_basic(t *testing.T) {
...
+	)
+
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&instance,
+		getOpenGaussInstanceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
```
Comment (niuzhenguo) [inline huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance_test.go]: better to add a specific precheck for opengauss, it costs too much

Comment (Lance52259) [reply huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance_test.go]: The precheck has been added:
```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccOpenGaussInstance_prepaid'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccOpenGaussInstance_prepaid -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_prepaid
=== PAUSE TestAccOpenGaussInstance_prepaid
=== CONT  TestAccOpenGaussInstance_prepaid
    acceptance.go:265: Do not allow expensive testing
--- SKIP: TestAccOpenGaussInstance_prepaid (0.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   0.060s
```

File: docs/resources/gaussdb_opengauss_instance.md [hunk 2]
```diff
@@ -163,6 +163,26 @@ The following arguments are supported:
 * `backup_strategy` - (Optional, List) Specifies the advanced backup policy.
   The [object](#opengauss_backup_strategy) structure is documented below.
 
+* `charging_mode` - (Optional, String, ForceNew) Specifies the charging mode of the vault.
+  The valid values are as follows:
+  + **prePaid**: the yearly/monthly billing mode.
+  + **postPaid**: the pay-per-use billing mode.
+
+  Changing this parameter will create a new resource.
```
Comment (Jason-Zhang9309) [inline docs/resources/gaussdb_opengauss_instance.md]: defaults to?

File: huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance_test.go [hunk 3]
```diff
@@ -79,6 +79,68 @@ func TestAccOpenGaussInstance_basic(t *testing.T) {
 	})
 }
 
+func TestAccOpenGaussInstance_prepaid(t *testing.T) {
+	var (
+		instance     instances.GaussDBInstance
+		resourceName = "huaweicloud_gaussdb_opengauss_instance.test"
+		rName        = acceptance.RandomAccResourceNameWithDash()
+		password     = fmt.Sprintf("%s@123", acctest.RandString(5))
+		newPassword  = fmt.Sprintf("%sUpdate@123", acctest.RandString(5))
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance_test.go]: you can use acceptance.RandomPassword()

File: huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go [hunk 4]
```diff
@@ -371,6 +376,7 @@ func buildOpenGaussInstanceCreateOpts(ctx context.Context, d *schema.ResourceDat
 		ConfigurationId:     d.Get("configuration_id").(string),
 		ShardingNum:         d.Get("sharding_num").(int),
 		CoordinatorNum:      d.Get("coordinator_num").(int),
+		ReplicaNum:          d.Get("replica_num").(int), // In prePaid payment, the replica number is required.
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go]: looks this parameter won't be empty, so this comment is unnecessary

Comment (Lance52259) [reply huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance.go]: I just wanted to explain why this parameter needs to be supported even the default value of replica number is fixed to three in API and schema.

File: docs/resources/gaussdb_opengauss_instance.md (hunk 2 above)
Comment (Lance52259) [reply docs/resources/gaussdb_opengauss_instance.md]: sure, defaults to 'postPaid'.

File: huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance_test.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/gaussdb/resource_huaweicloud_gaussdb_opengauss_instance_test.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccOpenGaussInstance_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccOpenGaussInstance_ -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_basic
=== PAUSE TestAccOpenGaussInstance_basic
=== RUN   TestAccOpenGaussInstance_prepaid
=== PAUSE TestAccOpenGaussInstance_prepaid
=== RUN   TestAccOpenGaussInstance_haModeCentralized
=== PAUSE TestAccOpenGaussInstance_haModeCentralized
=== CONT  TestAccOpenGaussInstance_basic
    acceptance.go:265: Do not allow expensive testing
=== CONT  TestAccOpenGaussInstance_haModeCentralized
    acceptance.go:265: Do not allow expensive testing
--- SKIP: TestAccOpenGaussInstance_basic (0.00s)
=== CONT  TestAccOpenGaussInstance_prepaid
    acceptance.go:265: Do not allow expensive testing
--- SKIP: TestAccOpenGaussInstance_prepaid (0.00s)
--- SKIP: TestAccOpenGaussInstance_haModeCentralized (0.00s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   0.064s
```

File: docs/resources/gaussdb_opengauss_instance.md [hunk 5]
```diff
@@ -163,6 +163,26 @@ The following arguments are supported:
 * `backup_strategy` - (Optional, List) Specifies the advanced backup policy.
   The [object](#opengauss_backup_strategy) structure is documented below.
 
+* `charging_mode` - (Optional, String, ForceNew) Specifies the charging mode of the vault. Defaults to **postPaid**.
```
Comment (niuzhenguo) [inline docs/resources/gaussdb_opengauss_instance.md]: s/valult/opengauss

File: docs/resources/gaussdb_opengauss_instance.md [hunk 6]
```diff
@@ -163,6 +163,26 @@ The following arguments are supported:
...
+  The valid values are as follows:
+  + **prePaid**: the yearly/monthly billing mode.
+  + **postPaid**: the pay-per-use billing mode.
+
+  Changing this parameter will create a new resource.
+
+* `period_unit` - (Optional, String, ForceNew) Specifies the charging period unit of the vault.
```
Comment (niuzhenguo) [inline docs/resources/gaussdb_opengauss_instance.md]: ditto

File: docs/resources/gaussdb_opengauss_instance.md [hunk 7]
```diff
@@ -163,6 +163,26 @@ The following arguments are supported:
...
+  Valid values are **month** and **year**. This parameter is mandatory if `charging_mode` is set to **prePaid**.
+  Changing this parameter will create a new resource.
+
+* `period` - (Optional, Int, ForceNew) Specifies the charging period of the vault.
```
Comment (niuzhenguo) [inline docs/resources/gaussdb_opengauss_instance.md]: ditto

File: docs/resources/gaussdb_opengauss_instance.md (hunk 5 above)
Comment (Lance52259) [reply docs/resources/gaussdb_opengauss_instance.md]: done.

**Comment by Lance52259** [comment]
It also works fine when the "IsAutoPay" field is true when the charge mode is "postPaid".
```
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb -v -run=TestAccOpenGaussInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccOpenGaussInstance_basic
=== PAUSE TestAccOpenGaussInstance_basic
=== CONT  TestAccOpenGaussInstance_basic
--- PASS: TestAccOpenGaussInstance_basic (4373.70s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb  4373.792s
```

**Pull request merged**